File: language-testing/fibonacci/fibonacci.py
#! /usr/bin/env python3

# Prints out numbers from the fibonacci sequence
# Example usage: ./fibonacci.py
# Example usage: ./fibonacci.py 42

from sys import argv

def fib(n):
    old_value = 1
    new_value = 0
    for _ in range(n):
        old_value, new_value = new_value, new_value + old_value
    return new_value

# If no arguments, print 20 fibs
# Otherwise we'll fib the first argument

if __name__ == "__main__":
    if len(argv) > 1:
        print(fib(int(argv[1])))
    else:
        print("\n".join("%s\t%s" % (fib(r), fib(r+10)) for r in range(10)))

File: language-testing/fibonacci/fibonacci.rb
#! /usr/bin/env ruby
# Prints out numbers from the fibonacci sequence
# Example usage: ./fibonacci.rb
# Example usage: ./fibonacci.rb 42

def fib(n)
  old_value = 1
  new_value = 0
  n.times { old_value, new_value = new_value, new_value + old_value }
  new_value
end

# If no arguments; run 20 fibs, else; fib the first argument:
if __FILE__ == $0
  if ARGV.length == 0
    (0...10).each { |i| puts "#{fib(i)}\t#{fib(i + 10)}" }
  else
    puts(fib(ARGV[0].to_i))
  end
end
